### PR TITLE
Improve babel config

### PR DESCRIPTION
**Pull request by pustovalov**
these modules:
```
@babel/plugin-syntax-dynamic-import
@babel/plugin-transform-destructuring
@babel/plugin-proposal-object-rest-spread
@babel/plugin-transform-regenerator
```

are already in `babel-preset-env`

https://github.com/babel/babel/blame/master/packages/babel-preset-env/src/available-plugins.js

current config: https://bit.ly/3czlbSQ

just preset-env: https://bit.ly/2XEIIO7

### Remove `forceAllTransforms`
https://babeljs.io/docs/en/babel-preset-env#forcealltransforms
target: `last 2 Chrome versions`

`forceAllTransforms false`: 59 output lines (https://bit.ly/3h7gPWL)

`forceAllTransforms true`:  183 output lines (https://bit.ly/377AAbT)

### Loose
https://babeljs.io/docs/en/babel-preset-env#loose
target: `defaults`

`Loose true`: 172 lines (https://bit.ly/378mMOq)

`loose false`: 182 lines (https://bit.ly/2XBVqwQ)

### Bugfixes
https://babeljs.io/docs/en/babel-preset-env#bugfixes
`Note: These optimizations will be enabled by default in Babel 8`

target: `last 2 Chrome versions, edge 16`

`bugfixes false`: 15 lines output (https://bit.ly/2UeYuwM)

`bugfixes true`: 6 output lines (https://bit.ly/2UeYuwM)

### Final output with all changes in this PR
target: `defaults`

https://bit.ly/378mMOq

#### Vendors bundle size
target `defaults`:
```
useBuiltIns: 'entry',
modules: false,
bugfixes: true,
loose: true,
corejs: 3,
```
`Size: 729.95 KB`

```
useBuiltIns: 'entry',
modules: false,
forceAllTransforms: true,
corejs: 3,
```
` Size: 746.48 KB`

### CODE REVIEW

File: lib/install/config/babel.config.js [hunk 1]
```diff
@@ -28,44 +28,31 @@ module.exports = function(api) {
...
+          bugfixes: true,
+          loose: true,
           exclude: ['transform-typeof-symbol']
         }
       ]
     ].filter(Boolean),
     plugins: [
       'babel-plugin-macros',
-      '@babel/plugin-syntax-dynamic-import',
       isTestEnv && 'babel-plugin-dynamic-import-node',
```
Comment (SkipKayhil) [inline lib/install/config/babel.config.js:42]: This should also be able to be removed, see https://github.com/facebook/create-react-app/pull/8353 and https://babeljs.io/blog/2020/01/11/7.8.0 and

**Comment by rossta** [comment]
Good catch with removing `forceAllTransforms`. Based on my reading of the docs, it looks like it's only useful when used with UglifyJS (not in Webpacker) or for environments that support only ES5 (e.g., for developers targeting IE11). Perhaps that last point should be documented in the README/Webpacker docs if it's to be removed from the default?

https://babeljs.io/docs/en/babel-preset-env#forcealltransforms

**Comment by pustovalov** [comment]
If you add to the target `ie 11` with `forceAllTransforms false` it will same output like with `forceAllTransforms true`

`target: last 2 Chrome versions`

`forceAllTransforms false`: 59 output lines (https://bit.ly/3h7gPWL)

`forceAllTransforms true`: 183 output lines (https://bit.ly/377AAbT)

`target: ie 11`
`forceAllTransforms false`: 183 output lines (https://bit.ly/2zjmiIU)

**Comment by gauravtiwari** [comment]
Thanks @pustovalov - this is lovely 🍰 

Could we document the target bit in the docs?

**Pull request merged**